safetensors/src/lib.rs

```diff
@@ -204,14 +206,19 @@ impl<'data> SafeTensors<'data> {
     /// The tensor returned is merely a view and the data is not owned by this
     /// structure.
     pub fn names(&self) -> Vec<&'_ String> {
-        self.metadata.0.iter().map(|(name, _)| name).collect()
+        self.metadata.1.iter().map(|(name, _)| name).collect()
+    }
+
+    /// Returns additional metadata passed along by the user.
+    pub fn get_metadata(&self) -> HashMap<String, String> {
+        self.metadata.0.clone()
```

Comment:
Here add to clone otherwise rust was complaining that the value was moved. Could maybe be avoided by making the filed public in the metadata (which would have to be public as well). I don't expect this to be a big object anyway.

Reply:
You could also return a reference to the map instead:

```rust
pub fn get_metadata(&self) -> &HashMap<String, String> {
        &self.metadata.0
```
Should work.

Since this struct already has an associated lifetime (the lifetime of the buffer) you might need more lifetime annotations to explicitly tell the compiler that both lifetimes are the same.

```rust
pub fn get_metadata(&'data self) -> &'data HashMap<String, String> {
        &self.metadata.0
```

Since `SafeTensors` cannot live more than `'data` , if you are handing out a reference to someone, then that reference cannot outlive YOU (and therefore neither the longest `'data`) .

In general the borrow checker has simple rules which elide the lifetimes, but it might not work here.